ui/components/multichain/network-list-menu/network-list-menu.js

```diff
@@ -303,15 +310,20 @@ export const NetworkListMenu = ({ onClose }) => {
         onClick={() => {
           dispatch(toggleNetworkMenu());
           dispatch(setActiveNetwork(network.providerType || network.id));
-
-          // If presently on and connected to a dapp, communicate a change to
-          // the dapp via silent switchEthereumChain that the network has
-          // changed due to user action
+          if (process.env.CHAIN_PERMISSIONS) {
```

Comment:
Can you share the context for what this does?  This file is rewritten in the default RPC work and I want to make sure this bit gets integrated correctly